10/root/usr/share/container-scripts/postgresql/common.sh

```diff
@@ -296,6 +296,11 @@ function wait_for_postgresql_master() {
 
 run_pgupgrade ()
 (
+  # Remove .pid file if the file persists after ugly shut down
+  if [ -f "$PGDATA/postmaster.pid" ] && ! ps -aux | awk '{print $11}' | grep -q postgres; then
```

Comment:
What do we want to check with this condition?

Is it if postgres process is running? If yes, could we use pgrep for that instead of ps?